### PR TITLE
feat(doc): how to incorporate static code tables in an API

### PR DESCRIPTION
**Pull request by gnito-org**

### CODE REVIEW

**Comment by dunglas** [comment]
Nice one! Couldn't you use the new `enum` type introduced in PHP 8.1 for this use case?

**Comment by gnito-org** [comment]
I have not looked at the new `enum` feature in depth. From cursory reading it appears that an `enum` is static. How would one instantiate it so that API Platform can treat is as just another entity, with an IRI?

**Comment by gnito-org** [comment]
@dunglas I added an `enum` section to the explanation.

File: core/static-code-tables.md [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+<?php
+// api/src/Entity/ItemType.php
+
+namespace App\Entity;
+
+use ApiPlatform\Core\Annotation\ApiProperty;
+use ApiPlatform\Core\Annotation\ApiResource;
+use Symfony\Component\Serializer\Annotation\Groups;
+
+#[ApiResource(
```
Comment (vincentchalamon) [inline core/static-code-tables.md:213]: Why creating an ApiResource on an enum?

**Pull request closed without merging**